graphql_compiler/schema/__init__.py

```diff
@@ -447,3 +451,29 @@ def check_for_nondefault_directive_names(directives):
         raise AssertionError(
             u"Unsupported directives found: {}".format(nondefault_directives_found)
         )
+
+
+def compute_schema_fingerprint(schema: GraphQLSchema) -> str:
+    """Compute a fingerprint compactly representing the data in the given schema.
+
+    The fingerprint is not sensitive to things like type or field order. This function is guaranteed
+    to be robust enough that if two GraphQLSchema have the same fingerprint, then they also
+    represent the same schema.
+    
+    Because of internal implementation changes, different versions of this library *may* produce
+    different fingerprints for the same schema. Since cross-version fingerprint stability
+    is an *explicit non-goal* here, changing a schema's fingerprint will not be considered 
+    a breaking change.
+
+    The fingerprint may not include information about directives at field definitions.
```

Comment:
This is way too specific. There are probably half a dozen other kinds of things that are also not covered (e.g. directives on types, on scalars, on extensions) and yet are not mentioned. Let's not actively mislead people by listing only one of the many caveats present here.

I'd keep it more vague and say that this fingerprint is done on a best-effort basis and has some known issues at the moment, and direct people to the discussion in this PR for more details.